python/ray/data/read_api.py

```diff
@@ -933,10 +933,10 @@ def read_json(
     ignore_missing_paths: bool = False,
     **arrow_json_args,
 ) -> Dataset:
-    """Creates a :class:`~ray.data.Dataset` from JSON files.
+    """Creates a :class:`~ray.data.Dataset` from JSON and JSONL files.
```

Comment:
PyArrow's `read_json` only support JSONL, not normal JSON.

> Currently only the line-delimited JSON format is supported.

```suggestion
    """Creates a :class:`~ray.data.Dataset` from JSONL files.
```

Reply:
I think it supports both JSONL and JSON format. Essentially JSON is just a line of JSONL.

Tried with the example in https://json.org/example.html:

```py
>>> import ray
>>> ds = ray.data.read_json("/Users/chengsu/try/json/input.json")
>>> ds
Dataset(
   num_blocks=20,
   num_rows=1,
   schema={
      glos...: struct<title: string, GlossDiv: struct<title: string, GlossList: struct<GlossEntry: struct<ID: string, SortAs: string, GlossTerm: string, Acronym: string, Abbrev: string, GlossDef: struct<para: string, GlossSeeAlso: list<item: string>>, GlossSee: string>>>>
   }
)
```

Reply:
We can document this: `For an input json file, the whole file will be read as one row. For an input jsonl file, each line will be read as one row. `

Reply:
Yes, updated.